### PR TITLE
feat:add video upload functionality

### PR DESCRIPTION
**Pull request by ShenDing1125**
add video upload functionality

### CODE REVIEW

**Comment by ShenDing1125** [comment]
[(3)【課程管理】可以切換課程簡介頁版型的功能](https://trello.com/c/RdOBtK6f/5315-3%E3%80%90%E8%AA%B2%E7%A8%8B%E7%AE%A1%E7%90%86%E3%80%91%E5%8F%AF%E4%BB%A5%E5%88%87%E6%8F%9B%E8%AA%B2%E7%A8%8B%E7%B0%A1%E4%BB%8B%E9%A0%81%E7%89%88%E5%9E%8B%E7%9A%84%E5%8A%9F%E8%83%BD)

File: src/pages/ProgramAdminPage/ProgramCoverForm.tsx [hunk 1]
```diff
@@ -104,18 +125,20 @@ const ProgramCoverForm: React.VFC<{
...
+    const uploadCoverUrl = !coverMedia
+      ? coverDefaultUrl
+      : validateFile(coverMedia.name) === 'image'
+      ? getImageSizedUrl(isUseOriginSizeCoverImage, coverMedia ? defaultUploadCoverUrl : coverDefaultUrl)
+      : defaultUploadCoverUrl
+
+    const defaultUploadMobileUrl = `https://${process.env.REACT_APP_S3_BUCKET}/program_covers/${appId}/${programId}/${mobileCoverId}`
+    const uploadMobileUrl = !coverMobileMedia
+      ? coverMobileUrl
+      : validateFile(coverMobileMedia.name) === 'image'
```
Comment (LoisChen68) [inline src/pages/ProgramAdminPage/ProgramCoverForm.tsx:138]: Using `coverMedia.type`.
Please try `console.log(coverMedia)` to see data  when uploading the file.

And whether `coverMobileUrl` should upload the video format needs to be confirmed.

File: src/pages/ProgramAdminPage/ProgramCoverForm.tsx [hunk 2]
```diff
@@ -129,11 +152,13 @@ const ProgramCoverForm: React.VFC<{
         coverDefaultUrl: uploadCoverUrl,
         coverMobileUrl: uploadMobileUrl,
         coverThumbnailUrl: uploadThumbnailUrl,
+        coverType: coverMedia ? validateFile(coverMedia.name) : 'image',
+        mobileCoverType: coverMobileMedia ? validateFile(coverMobileMedia.name) : 'image',
```
Comment (LoisChen68) [inline src/pages/ProgramAdminPage/ProgramCoverForm.tsx:156]: Using `coverMedia.type`.
Please try `console.log(coverMedia)` to see data  when uploading the file.

File: src/pages/ProgramAdminPage/ProgramCoverForm.tsx [hunk 3]
```diff
@@ -175,14 +203,16 @@ const ProgramCoverForm: React.VFC<{
...
               onChange={file => {
-                setCoverImage(file)
+                setCoverMedia(file)
                 setIsUseOriginSizeCoverImage(false)
               }}
+              customName={formatMessage(commonMessages.ui.uploadMedia)}
+              customImg={!isShowOriginSizeCover ? EmptyVideoCover : undefined}
             />
-            {(!isEmpty(coverDefaultUrl) || coverImage) && (
+            {isShowOriginSizeCover && (
```
Comment (LoisChen68) [inline src/pages/ProgramAdminPage/ProgramCoverForm.tsx:215]: If it is confirmed that this `checkbox` does not need to be displayed in the video format, you can using `coverMidea.type` without use setState.

File: src/pages/ProgramAdminPage/ProgramCoverForm.tsx [hunk 4]
```diff
@@ -55,10 +66,20 @@ const ProgramCoverForm: React.VFC<{
...
+  const [isShowOriginSizeCover, setIsShowOriginSizeCover] = useState(!isEmpty(coverDefaultUrl) && coverType === 'image')
+  const [isShowOriginSizeMobileCover, setIsShowOriginSizeMobileCover] = useState(
+    !isEmpty(coverMobileUrl) && mobileCoverType === 'image',
+  )
+
+  useEffect(() => {
+    if (coverMedia) setIsShowOriginSizeCover(validateFile(coverMedia.name) === 'image')
+    if (coverMobileMedia) setIsShowOriginSizeMobileCover(validateFile(coverMobileMedia.name) === 'image')
+  }, [coverMedia, coverMobileMedia])
+
```
Comment (LoisChen68) [inline src/pages/ProgramAdminPage/ProgramCoverForm.tsx:82]: Use `coverMedia.type` to whether to display, don’t need this `useEffect`.

**Review by LoisChen68 — CHANGES_REQUESTED** [review]
Please modify code or reply to comment.

**Pull request merged**